engine/src/main/java/org/hibernate/search/query/dsl/impl/ConnectedMultiFieldsPlainTextQueryBuilder.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+
+package org.hibernate.search.query.dsl.impl;
+
+import java.util.List;
+
+import org.apache.lucene.search.BooleanClause.Occur;
+import org.apache.lucene.search.Query;
+import org.hibernate.search.analyzer.impl.LuceneAnalyzerReference;
+import org.hibernate.search.analyzer.impl.RemoteAnalyzerReference;
+import org.hibernate.search.query.dsl.PlainTextTermination;
+import org.hibernate.search.util.StringHelper;
+import org.hibernate.search.util.logging.impl.Log;
+import org.hibernate.search.util.logging.impl.LoggerFactory;
+
+/**
+ * @author Guillaume Smet
+ */
+public class ConnectedMultiFieldsPlainTextQueryBuilder implements PlainTextTermination {
+
+	private static final Log LOG = LoggerFactory.make();
+
+	private final String searchText;
+	private final QueryCustomizer queryCustomizer;
+	private final List<FieldsContext> fieldsContexts;
+	private final QueryBuildingContext queryContext;
+
+	private final boolean defaultOperatorIsAnd;
+
+	public ConnectedMultiFieldsPlainTextQueryBuilder(String searchText,
+			List<FieldsContext> fieldsContexts,
+			boolean defaultOperatorIsAnd,
+			QueryCustomizer queryCustomizer,
+			QueryBuildingContext queryContext) {
+		this.searchText = searchText;
+		this.queryContext = queryContext;
+		this.queryCustomizer = queryCustomizer;
+		this.fieldsContexts = fieldsContexts;
+
+		this.defaultOperatorIsAnd = defaultOperatorIsAnd;
+	}
+
+	@Override
+	public Query createQuery() {
+		if ( StringHelper.isEmpty( searchText ) ) {
+			throw LOG.simpleQueryParserDoesNotSupportEmptyQueries();
+		}
+
+		Query query;
+
+		if ( queryContext.getQueryAnalyzerReference().is( RemoteAnalyzerReference.class ) ) {
+			// we use the analyzers defined in the schema as it is not possible to define per field schema here
+			RemoteSimpleQueryStringQuery.Builder builder = new RemoteSimpleQueryStringQuery.Builder()
+					.query( searchText )
+					.defaultOperatorIsAnd( defaultOperatorIsAnd );
```

Comment:
We don't pass the reference to the analyzer here... If I understand correctly, this is because we can't define per-field analyzers in Elasticsearch's `simple_query_string`. We only have one global `analyzer` parameter.

Maybe this should be documented? Users may be surprised when they write:

```
searchFactory.buildQueryBuilder().forEntity( MyEntity.class ).overridesForField( "foo", "someQueryAnalyzer" ).get()
    .text().forField( "foo" ).matching( "blah blah" ).createQuery();
```

... and the resulting query doesn't use the overridden analyzer.

Maybe we should also support the case where there's only one field, and throw an error if there are multiple fields with different analyzers?